src/poly/cse.rs

```diff
@@ -0,0 +1,83 @@
+use rand_chacha::{rand_core::SeedableRng, ChaCha20Rng};
+
+use crate::field::Field;
+
+use std::{collections::HashSet, fmt::Debug, hash::Hash};
+
+use super::{Expr, HashResult, VarAssignments};
+
+/// Common Subexpression Elimination - takes a collection of expr
+pub fn cse<F: Field + Hash, V: Debug + Clone + Eq + Hash>(
+    exprs: Vec<Expr<F, V, ()>>,
```

Comment:
I think we can pass here the queriables, because the caller will have the list of all queriables and it will be more efficient than calling get_queries in all the expr

Reply:
Can we use https://github.com/privacy-scaling-explorations/chiquito/blob/93282e00b925fd3d61ea32c0f48fd8385f385b51/src/sbpir/mod.rs#L28C9-L28C20 directly instead of the queriables?

Reply:
The annotations? We need to know, if they are internal or forward. You could either pass queriables, or an array of internals and another of forward